api/src/pcapi/core/finance/repository.py

```diff
@@ -124,6 +124,7 @@ def has_active_or_future_custom_reimbursement_rule(offer: offers_models.Offer) -
     query = models.CustomReimbursementRule.query.filter(
         models.CustomReimbursementRule.offerId == offer.id,
         models.CustomReimbursementRule.timespan.overlaps(timespan),
+        models.CustomReimbursementRule.amount.is_not(None),
```

Comment:
Il ne devrait pas être nécessaire d'ajouter ce filtre. En prod, on a forcément un `amount` si on a `offerId`. On devrait d'ailleurs avoir une contrainte en base de données : `CHECK ("offerId" is not null and amount is not null) or ("offererId" is not null and rate is not null)`. Je me note de l'ajouter.

Est-ce que tu ajoutes ce filtre dans le code parce qu'il y a un bug quelque part **en prod** ? Si l'erreur est dans un test, c'est peut-être parce que la factory est incorrecte ?

Reply:
J'ai rajouté ce filtre parce que quand on crée des stocks pour une offre qui n'est pas encore créée, on a `offer.id = None` donc cette query renvoit les `CustomReimbursementRule` associé à des offerer

Il est sans doute plus naturel de remplacer `amount.is_not(None)` par `offerId.is_not(None)`, qu'en penses-tu ?

Reply:
problème contourné en flushant manuellement la session sqlalchemy